client/js/lib/prosemirror/plugins.js

```diff
@@ -336,39 +336,57 @@ const rangeTracker = (rangeTrackerKey, schema, rangeChangeCallback = () => {}) =
  */
 const rangeCreator = (pluginKey, rangeEditingKey) => {
   let tippy = null
+
+  const handleGlobalMouseup = (view, e) => {
+    const { state } = view
+    const { selection } = state
+    const { empty, from, to, $anchor, $head } = selection
+
+    if (empty) {
+      tippy?.destroy()
+      tippy = null
+      return
+    }
+
+    const lastClick = view.lastClick || view.input.lastClick
+    const hasClickLocationChanged = (e.clientX !== lastClick.x || e.clientY !== lastClick.y)
+
+    if (hasClickLocationChanged) {
+      const existingRanges = rangeEditingKey.getState(state)
+      let positionsCovered = []
+      Object.values(existingRanges).forEach(({ from, to }) => positionsCovered.push(...range(from, to)))
+      const selectedPositions = new Set(range(from, to))
+      positionsCovered = new Set(positionsCovered)
+      const isFullyCovered = isSuperset(positionsCovered, selectedPositions)
+
+      if (isFullyCovered) {
```

Comment:
I would like if you add a comment something like
/**
 * Prevent showing the tooltip if the user's selection is already fully covered by the selection before.
 * This avoids redundant actions like trying to create a new annotation inside an already annotated area.
 * The user can still select text within annotated ranges, but no tooltip will be shown in that case.
 */